From 3b6ddd1dd518fc241482df16e68e0d0b21d4a91a Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Thu, 28 Apr 2022 14:35:47 +0300
Subject: [PATCH] [SW-82716] gaudi3: add pcie device id api

One of OOB command which is send while firmware update
is getting pcie device id. This patch add routine
which will be called from indirect i2c command.

Change-Id: I8d5675331d8b3d0b6d34f66c32fd743507e70385
---

diff --git a/arch/gaudi3/arc/common.c b/arch/gaudi3/arc/common.c
index 0beea1c..b1bb431 100644
--- a/arch/gaudi3/arc/common.c
+++ b/arch/gaudi3/arc/common.c
@@ -880,3 +880,18 @@
 {
 	arcboot_rom_set_spi_tpm_cs_gpio(cs_en);
 }
+
+uint32_t arcboot_rom_get_device_id(void)
+{
+	if (arcboot_rom_global.config_fuse & EFUSE_SYSCFG_SECURITY_ENABLE_MASK) {
+		if (arcboot_rom_global.sku)
+			return PCI_IDS_GAUDI3_SINGLE_DIE_SEC;
+		else
+			return PCI_IDS_GAUDI3_SEC;
+	} else {
+		if (arcboot_rom_global.sku)
+			return PCI_IDS_GAUDI3_SINGLE_DIE;
+		else
+			return PCI_IDS_GAUDI3;
+	}
+}
diff --git a/include/arcboot.h b/include/arcboot.h
index 3871ab3..fd6fc56 100644
--- a/include/arcboot.h
+++ b/include/arcboot.h
@@ -284,6 +284,7 @@
 uint32_t arcboot_error_read_and_clr(uint32_t *major, uint32_t *minor);
 bool arcboot_error_inform(uint32_t major, uint32_t minor);
 bool arcboot_error_is_pending(void);
+uint32_t arcboot_rom_get_device_id(void);
 
 #define CLEAR_DMA_SECURITY_PROGRESS 0x3
 
